NN_training_example.py

```diff
@@ -1,53 +1,69 @@
 import numpy as np
+
 from light_prop.algorithms import NNTrainer
 from light_prop.calculations import get_gaussian_distribution
 from light_prop.lightfield import LightField
 from light_prop.propagation.params import PropagationParams
-from light_prop.visualisation import GeneratePropagationPlot
-from light_prop.calculations import gaussian
+from light_prop.visualisation import GeneratePropagationPlot, PlotTypes
 
 if __name__ == "__main__":
     params = PropagationParams.get_example_propagation_data()
 
-    #Choose proper propagation parameters
-    params.sigma = 2
+    # Choose proper propagation parameters
+    params.beam_diameter = 2
     params.matrix_size = 128
     params.pixel = 0.5
 
-
-    #Define target optical field and input amplitude
+    # Define target optical field and input amplitude
+    # In this example a simple focusing from wider Gaussian beam to the thinner one
     x0 = 0
     y0 = 0
     target = get_gaussian_distribution(params, x0, y0)
-    params.sigma = 15
+    params.beam_diameter = 15
     amp = get_gaussian_distribution(params, 0, 0)
     phase = np.array(
         [[0 for x in
           np.arange(-params.matrix_size / 2, params.matrix_size / 2) * params.pixel] for y in
          np.arange(-params.matrix_size / 2, params.matrix_size / 2) * params.pixel])
 
-    #Build NNTrainer
+    # Build NNTrainer
     NN = NNTrainer(params)
 
     # Run NN optimization
-    trained_model = NN.optimize(LightField(amp, phase), LightField(target, phase), 100)
+    # Please try running different numbers of epochs (last parameter)
```

Comment:
Discrepancy, `epochs` = `iterations`? If so I'd align to `iteration`. In additon imo `(last parameter)` remark is not needed since we have it present as named param.